### ISSUE TITLE
NonLinear SMV 

### ISSUE THREAD

**Issue by ozgurgundogan** [nfmcclure/tensorflow_cookbook#7]
Hi ,
you are using formulas given below to calculate your accuracy.

prediction_output = tf.matmul(tf.mul(tf.transpose(y_target),b), pred_kernel)
prediction = tf.sign(prediction_output-tf.reduce_mean(prediction_output))
accuracy = tf.reduce_mean(tf.cast(tf.equal(tf.squeeze(prediction), tf.squeeze(y_target)), tf.float32))

you are using directly y_target (ground truth) to make a prediction and then comparing the prediction with y_target. 
Firstly I dont understand what is going on there. I mean "predicting by using target value" sounds meaningless a little bit. Probably I am missing something, but I couldnt find what it is.

At the end of the day, if I want to predict output when test input data are applied without knowing output how can I predict my outputs ? 

By the way, there is a line in the code 

model_output = tf.matmul(b, my_kernel)

But you dont use model_output anywhere in your code.

Thanks.

**Comment by nfmcclure** [comment]
Hi @ozgurgundogan, thanks for bringing this to my attention.  There was some extra code lines in there from when I was exploring how to write this code.  I cleaned up the code a bit.  But I also added my best quick explanation in an Jupyter notebook here: https://github.com/nfmcclure/tensorflow_cookbook/blob/master/04_Support_Vector_Machines/05_Implementing_Nonlinear_SVMs/05_nonlinear_svm.ipynb

For better explanations, I would recommend reading these links:
http://cs229.stanford.edu/notes/cs229-notes3.pdf
http://www.robots.ox.ac.uk/~az/lectures/ml/lect3.pdf

As a note, there needs to be the x and y points in any prediction you do because it's the x and y points that create the SVM margins.  If you do some exploring with pred_kernel and my_kernel variable outputs, you'll see which non-zero values in the arrays indicate which data points are the support vectors.

I'll leave this issue open for a while if you want more clarification, otherwise I'll close it in about 48 hours.  Thanks!
